cli/ops/fs.rs

```diff
@@ -84,18 +99,29 @@ fn op_open(
       state.check_write(&path)?;
     }
 
+    if gave_mode && !(options.create || options.create_new) {
+      return Err(OpError::type_error(
+        "specified mode without allowing file creation".to_string(),
+      ));
+    }
+
     open_options
       .read(options.read)
       .create(options.create)
       .write(options.write)
       .truncate(options.truncate)
       .append(options.append)
       .create_new(options.create_new);
-  } else if let Some(mode) = args.mode {
-    let mode = mode.as_ref();
-    match mode {
+  } else if let Some(open_mode) = args.open_mode {
+    let open_mode = open_mode.as_ref();
+    match open_mode {
       "r" => {
         state.check_read(&path)?;
+        if gave_mode {
+          return Err(OpError::type_error(
+            "specified mode for read-only open".to_string(),
+          ));
+        }
```

Comment:
Unnecessary. Let lower level code handle it.

Reply:
If I remove this check, Rust raises no error for `Deno.openSync(existing_filepath, { mode:0o600, read:true, write:false});` ... but this would also be caught by the previous error-check, so that one may be enough. (If one attempts to open with `{create: true, write: false}`, we already catch that and report an error in `cli/js/files.ts`.)

I was thinking that supplying a `mode` option to `open` or `openSync` when neither `create` nor `createNew` was true was most likely to be a programming error, and so it'd be useful to throw, rather than just silently ignoring the `mode` option. But if you think the latter is better, it's easy to implement. I'll push that change. If in light of this info you think differently, you can just revert it.